DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h

```diff
@@ -1,244 +1,297 @@
+#ifndef L1_TRACK_TRIGGER_TRACK_WORD_H
+#define L1_TRACK_TRIGGER_TRACK_WORD_H
+
 ////////
 //
 // class to store the 96-bit track word produced by the L1 Track Trigger.  Intended to be inherited by L1 TTTrack.
 // packing scheme given below.
 //
-// author: Mike Hildreth
-// date:   April 9, 2019
+// author:      Mike Hildreth
+// modified by: Alexx Perloff
+// created:     April 9, 2019
+// modified:    March 9, 2021
 //
 ///////
 
-#ifndef L1_TRACK_TRIGGER_TRACK_WORD_H
-#define L1_TRACK_TRIGGER_TRACK_WORD_H
+#include "DataFormats/GeometryVector/interface/GlobalVector.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+
+#include <ap_int.h>
 
 #include <iostream>
 #include <string>
-#include "DataFormats/GeometryVector/interface/GlobalVector.h"
-#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include <vector>
 
 class TTTrack_TrackWord {
 public:
-  // Constructors:
+  // ----------constants, enums and typedefs ---------
+  enum TrackBitWidths {
```

Comment:
IIUC these constants define the actual format of the data.
In this case, does it make sense to make them configurable ?

Reply:
I count 52 constants here, though yes, some are derived from others. (1) I think @fwyzard is correct in that these define the data format and don't need to be easily configurable. (2) Several of the constants, like the `TrackBitWidths` and the step sizes, need to be available at compile time. At least, as the code is written right now. This would, as far as I'm aware, make it impossible to pass these in from a python cfg during runtime.

Reply:
> > Given that the digitized track parameters are derived from the floating point ones, do we actually need to store them as an EDProduct at all? Can't they be obtained on the fly? So perhaps better if TTTrack only contains floating point numbers. Would have added advantage that it could be redigitized if digitisation params change.
> 
> If we don't store them as EDProducts then the L1GTTInputProducer will need to be re-run for ever module which needs q/R converted to pt and tan(lambda) converted to eta. This way you just need to run that producer once. Also, aren't the digitized values eventually going to come from an emulator and this is just a placeholder solution? I'm not involved in that level of planning, so I don't know the answer.

The data formats have changed a few times in the past year, and may continue to do so. If we did the digitisation in a separate EDProducer, that consumed floating point info from TTTrack & wrote an EDProduct TTTrackWord, this could be rerun if the format changes. (N.B. If we don't do this, we'd need to rerun the whole L1 tracking sequence). And it the digi params could be passed to it via python. If all the helix params were packed into unsigned int (or similar), the data format would not change if the digi params did.

The L1 track sequence will eventually (~1 year) write digitized info directly, but this may be written to a separate EDProduct to TTTrack.